### PR TITLE
feat(atomic): add error handling for atomic-commerce queries

### PR DESCRIPTION
**Pull request by olamothe**
* Refactor `QueryErrorCommon` into multiple functional components
* Create new `atomic-commerce-query-error` component

For now it does not handle any commerce API specific errors. But it should at least have the "catch all" which is "Something went wrong" with a "Show more" button
https://coveord.atlassian.net/browse/KIT-3034

### CODE REVIEW

**Review by louis-bompart — APPROVED** [review]
Smooth 👌

File: packages/atomic/src/components/commerce/atomic-commerce-query-error/atomic-commerce-query-error.tsx [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+import {QueryErrorGuard} from '../../common/query-error/guard';
+import {QueryErrorIcon} from '../../common/query-error/icon';
+import {QueryErrorLink} from '../../common/query-error/link';
+import {QueryErrorShowMore} from '../../common/query-error/show-more';
+import {QueryErrorTitle} from '../../common/query-error/title';
+import {getAriaMessageFromErrorType} from '../../common/query-error/utils';
+import {CommerceBindings} from '../atomic-commerce-interface/atomic-commerce-interface';
+
+/**
+ * The `atomic-commerce-query-error` component handles fatal errors when performing a query on the Commerce API. When the error is known, it displays a link to relevant documentation link for debugging purposes. When the error is unknown, it displays a small text area with the JSON content of the error.
```
Comment (fbeaudoincoveo) [inline packages/atomic/src/components/commerce/atomic-commerce-query-error/atomic-commerce-query-error.tsx:29]:
```suggestion
 * The `atomic-commerce-query-error` component handles fatal errors when performing a query on the Commerce API. When the error is known, it displays a link to relevant documentation for debugging purposes. When the error is unknown, it displays a small text area with the JSON content of the error.
```

File: packages/atomic/src/components/commerce/atomic-commerce-query-error/atomic-commerce-query-error.tsx [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+ *
+ * @part icon - The svg related to the error.
```
Comment (fbeaudoincoveo) [inline packages/atomic/src/components/commerce/atomic-commerce-query-error/atomic-commerce-query-error.tsx:31]:
```suggestion
 * @part icon - The SVG related to the error.
```

**Review by fbeaudoincoveo — APPROVED** [review]

File: packages/atomic/src/components/common/query-error/link.tsx [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+      case 'NoEndpointsException':
+        return 'https://docs.coveo.com/en/mcc80216';
+      case 'InvalidTokenException':
+        return 'https://docs.coveo.com/en/102';
+      case 'OrganizationIsPausedException':
+        return 'https://docs.coveo.com/l6af0467';
+      default:
+        return null;
+    }
+  };
```
Comment (dmgauthier) [inline packages/atomic/src/components/common/query-error/link.tsx:25]: Small detail, This function will be recreated each time the component renders, I personally would declare it outside the component with the errorType as a param or wrap it in a memo with the errorType as a dependency.

**Review by dmgauthier — APPROVED** [review]

**Pull request merged**